### DISCUSSION TITLE
Session management and audio recording

### DISCUSSION THREAD

**Discussion by Aakashagr** [flet-dev/flet#569] [Q&A] [answered]
Hello, 

I am planning to build a few puzzle games but have no prior experience with app development (although I have used python extensively for data analysis). 
Can you please share any example of how to save user interaction and retrieve it back when the user logs in again? For example, save the to-do list when I reopen the app again. I assume it is related to session storage but I am looking for a more detailed example of its implementation. 

Also, is there functionality in flet to record audio and upload it to the server?

Thanks a lot
Aakash

**Comment by Aakashagr** [comment, marked as answer]
I found a nice project that includes both frontend and backend using flet and fastapi
https://github.com/kareemmahlees/Team-Manager

**Reply by FeodorFitsner** [reply]
So, what is the benefit of FastAPI? Is it for user management functionality primarily?

**Reply by Aakashagr** [reply]
Yeah, that's right.  For my project, it would help me access the content from the database and also upload users' interactions within the app.

**Reply by FeodorFitsner** [reply]
OK, but it's kind of overhead to call a "backend" service from Flet via HTTP/JSON when Flet itself is already a "backend" service.
You can talk to a database or a native API directly. Does FastAPI provide a direct access to its user management API?

**Reply by Aakashagr** [reply]
Oh, I see. As I have no experience in app development, I am trying to learn through example codes and tutorials. It would be very helpful if you could provide a short tutorial on that. I suppose this package might be related to what you are asking: https://fastapi-users.github.io/fastapi-users/10.2/

(*not sure if it answers your question as I don't fully understand it yet)

**Reply by FeodorFitsner** [reply]
Will take a look!
